### PR TITLE
Basic bazel support.

### PR DESCRIPTION
**Pull request by Nurdok**
It's possible now to add a BUILD file with the following rule:

```
load("//:lumi.bzl", "lumi_binary")

lumi_binary(
    name = "hello-world",
    srcs = ["hello-world.5.lm"],
    deps = [],
)
```

and build it with

```
bazel build //my_path:hello-world
```

This will generate the executable in

```
bazel-bin/my_path/hello-world
```

(with `.exe` on Windows).

`lumi_binary` runs only with the latest lumi compiler version. It's possible to add a `compiler` flag if needed.

For a future PR: add `lumi_test`.

### CODE REVIEW

**Comment by Nurdok** [comment]
Note to self: roll back or forward with include file change in `lumi.c` and `tl5-compiler.c`, fix `.bzl` and `BUILD` indentations.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,3 @@
-build
```
Comment (meircif) [inline .gitignore]: Why? I do want some ignored build folder

File: lumi.bzl [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	),
+	"deps": attr.label_list(),
+    },
+    implementation = _generated_c_impl,
+    outputs = {"out": "%{name}.c"},
+    output_to_genfiles = True,
+)
+
+
+def lumi_binary(name, srcs, deps, **kwargs):
```
Comment (meircif) [inline lumi.bzl]: Currently Lumi may generate some unused variables and labels in the C code,
maybe add `-Wno-unused-label  -Wno-unused-variable` flags to the C compilation (and `-Wunused-but-set-variable` when using GCC)? so there will be no ugly warnings when compiling...

Comment (meircif) [reply lumi.bzl]: also ` -Wno-missing-braces` (and `-Wno-self-assign` when using `clang`)

File: .gitignore [hunk 3]
```diff
@@ -28,3 +29,6 @@ CMakeLists.txt
 
 # Atom files
 .tags
+
+# Bazel
+bazel-*/
```
Comment (meircif) [inline .gitignore]: This doesn't ignore the symlinks created by bazel in my Linux, maybe it should be `bazel-*` only (with no `/`)

File: lumi.bzl (hunk 2 above)
Comment (Nurdok) [reply lumi.bzl]: This doesn't work for me on Windows with Blaze, for now. I think we can merge this and fix later if needed (it's a bit spammy, but it's not that bad).

Comment (Nurdok) [reply lumi.bzl]: * Bazel

File: lumi.bzl [hunk 4]
```diff
@@ -48,5 +48,10 @@ def lumi_binary(name, srcs, deps, **kwargs):
         name = name,
         srcs = [":" + generated_name],
         deps = [],
+	copts = [
+            "-Wno-unused-label",
+            "-Wno-unused-variable",
+            "-Wno-unused-but-set-variable",
```
Comment (meircif) [inline lumi.bzl]: Probably not working for clang

**Pull request merged**